### PR TITLE
replacing docker_env_metadata_whitelist with env_metadata_whitelist instead of removing it 

### PR DESCRIPTION
**Pull request by zouyee**
#### What type of PR is this?

/kind regression
#### What this PR does / why we need it:
In order to remove dockershim, the Docker-related command lines built-in cadvisor, such as `docker_env_metadata_whitelist`, were removed in PR #97252. However, the command lines related to containerd were retained. Therefore, it is worth considering replacing `docker_env_metadata_whitelist` with `containerd_env_metadata_whitelist` instead of removing it directly. This approach ensures compatibility after runtime switching and maintains management stability.

Although cadvisor provides the `env_metadata_whitelist` command line, it currently supports only two runtimes. Directly replacing it might introduce more vendor code, increasing complexity. Hence, opting to replace it with `containerd_env_metadata_whitelist` is a more cautious and prudent choice.

This method maintains stability at the code level without introducing excessive external dependencies, ensuring system reliability and maintainability.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
replacing docker_env_metadata_whitelist with containerd_env_metadata_whitelist instead of removing it
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
@zouyee - can you not switch to `env_metadata_whitelist` ( https://github.com/google/cadvisor/blob/688835d77b199a4e95497b7d7d7b3580d5b03860/cmd/cadvisor.go#L68 )

**Comment by zouyee** [comment]
> Although cadvisor provides the `env_metadata_whitelist` command line, it currently supports only two runtimes. Directly replacing it might introduce more vendor code, increasing complexity. Hence, opting to replace it with `containerd_env_metadata_whitelist` is a more cautious and prudent choice.


Although cadvisor provides the `env_metadata_whitelist` command line, it currently supports only two runtimes(`docker` and `containerd`). Directly replacing it might introduce more vendor code, increasing complexity(see https://github.com/google/cadvisor/issues/2442). Hence, opting to replace it with `containerd_env_metadata_whitelist` is a more cautious and prudent choice.

**Comment by zouyee** [comment]
@dims PTAL

File: pkg/kubelet/cadvisor/cadvisor_linux.go [hunk 1]
```diff
@@ -101,9 +104,8 @@ func New(imageFsInfoProvider ImageFsInfoProvider, rootPath string, cgroupRoots [
 		Interval:     &duration,
 		AllowDynamic: pointer.Bool(allowDynamicHousekeeping),
 	}
-
 	// Create the cAdvisor container manager.
-	m, err := manager.New(memory.New(statsCacheDuration, nil), sysFs, housekeepingConfig, includedMetrics, http.DefaultClient, cgroupRoots, nil /* containerEnvMetadataWhiteList */, "" /* perfEventsFile */, time.Duration(0) /*resctrlInterval*/)
+	m, err := manager.New(memory.New(statsCacheDuration, nil), sysFs, housekeepingConfig, includedMetrics, http.DefaultClient, cgroupRoots, strings.Split(*envMetadataWhiteList, ",") /* containerEnvMetadataWhiteList */, "" /* perfEventsFile */, time.Duration(0) /*resctrlInterval*/)
```
Comment (bart0sh) [inline pkg/kubelet/cadvisor/cadvisor_linux.go:108]: Just wondering if `containerEnvMetadataWhiteList` is a better name for this variable? It looks like this name was actually used here, but commented out.

Comment (zouyee) [reply pkg/kubelet/cadvisor/cadvisor_linux.go:108]: envMetadataWhiteList here is copied from https://github.com/google/cadvisor/blob/688835d77b199a4e95497b7d7d7b3580d5b03860/cmd/cadvisor.go#L68.

**Pull request closed without merging**